Algorithm.CSharp/OrderImmutabilityRegressionAlgorithm.cs

```diff
@@ -72,37 +85,56 @@ public override void OnOrderEvent(OrderEvent orderEvent)
                 throw new Exception("Orders should be clones, hence not equal!");
             }
 
+            // Try and manipulate orderV2 using the only external accessor BrokerID, since we
+            // are changing a clone the BrokerIDs should not be the same
+            orderV2.BrokerId.Add("FAKE BROKER ID");
+            var orderV3 = Transactions.GetOrderById(orderEvent.OrderId);
+
+            if (orderV2.BrokerId == orderV3.BrokerId)
```

Comment:
Here it's actually checking for reference equality, we should verify the lists items, linq has a `SequenceEqual` extension

Reply:
Got it!